src/pokedex.c

```diff
@@ -5084,74 +5084,24 @@ static void sub_80927F0(u8 taskId)
         DestroyTask(taskId);
 }
 
-#ifdef NONMATCHING
 void sub_8092810(u8 a, u8 b, u8 c, u8 d)
 {
-    u16 i;
+    u16 i, j;
 
     for (i = 0; i < d; i++)
     {
-        ((u16 *)VRAM)[15 * 0x400 + c * 32 + i + b] &= 0xFFF;
-        ((u16 *)VRAM)[15 * 0x400 + c * 32 + i + b] |= a << 12;
+        j = *(u16 *)(VRAM + 15 * 0x800 + c * 64 + (b + i)*2);
+        j &= 0xFFF;
+        j |= a << 12;
+
+        *(u16 *)(VRAM + 15 * 0x800 + c * 64 + (b + i)*2) = j;
```

Comment:
This looks macro-like

Reply:
Should I make a macro?

Reply:
I don't think a macro is needed because macros are for reuse. this is the only time I do this.